CMakeLists.txt

```diff
@@ -240,18 +240,19 @@ config_summary()
 
 
 #***************************** Source files ****************************
-add_subdirectory( knowhere/utils )
-add_subdirectory( thirdparty )
-add_subdirectory( knowhere )
-
-if ( KNOWHERE_BUILD_TESTS )
-    if ( BUILD_COVERAGE STREQUAL "ON" )
+if ( BUILD_COVERAGE STREQUAL "ON" )
         append_flags( CMAKE_CXX_FLAGS
```

Comment:
this flag should be appended only when both "BUILD_COVERAGE" and "KNOWHERE_BUILD_TESTS" enabled

Reply:
"BUILD_COVERAGE" should be isolated with unittest because other interface such as python tests also need the code coverage statistics